src/lib/Hydra/Controller/User.pm

```diff
@@ -92,14 +104,16 @@ sub doLDAPLogin {
 }
 
 sub doEmailLogin {
-    my ($self, $c, $type, $email, $fullName) = @_;
+    my ($self, $c, $type, $email, $fullName, $username) = @_;
 
     die "No email address provided.\n" unless defined $email;
 
     # Be paranoid about the email address format, since we do use it
     # in URLs.
     die "Illegal email address.\n" unless $email =~ /^[a-zA-Z0-9\.\-\_]+@[a-zA-Z0-9\.\-\_]+$/;
```

Comment:
Unrelated: this is the absolute worst because it excludes plus-emails

Reply:
Yes.